test/unit/field-locale.spec.js

```diff
@@ -99,49 +102,42 @@ describe('FieldLocale', () => {
       return field.onValueChanged(noop)
     })
 
-    it('calls handler immediately on attach with initial value of field', () => {
-      const spy = sinon.spy()
-      field.onValueChanged(spy)
-      sinon.assert.calledOnce(spy)
-      sinon.assert.calledWithExactly(spy, info.value)
-    })
-
-    describe('New value equals current value', () => {
-      it('does not dispatch new value', () => {
-        const oldValue = field.getValue()
-        const cb = sinon.spy()
+    it('calls handler immediately on attach with most recently received value', () => {
+      channelStub.receiveMethod('valueChanged', ['some-field', 'en-US', 'VAL1'])
+      const spy1 = sinon.spy()
+      field.onValueChanged(spy1)
+      sinon.assert.calledOnce(spy1)
+      sinon.assert.calledWithExactly(spy1, 'VAL1')
 
-        field.onValueChanged(cb)
-        sinon.assert.calledOnce(cb)
-        valueChangedHandler(field.id, defaultLocale, oldValue)
-        sinon.assert.calledOnce(cb)
-      })
+      channelStub.receiveMethod('valueChanged', ['some-field', 'en-US', 'VAL2'])
+      const spy2 = sinon.spy()
+      field.onValueChanged(spy2)
+      sinon.assert.calledOnce(spy2)
+      sinon.assert.calledWithExactly(spy2, 'VAL2')
     })
 
-    describe('targeted at another field’s id', () => {
-      it('does not update the value', () => {
-        const oldValue = field.getValue()
-        valueChangedHandler(`${field.id}-other-id`, defaultLocale, newValue)
+    it('calls handler when value change is received', () => {
+      const spy = sinon.spy()
+      field.onValueChanged(spy)
+      spy.reset()
```

Comment:
The thing is, if you initialize it and then don't use it in `onValueChanged` like I said, then you don't need to reset it since it just got initialized (same state).

Or am I missing something?